R/test-helper.R

```diff
@@ -2,8 +2,11 @@
 prepare_example_db <- function(db_name, dir = getwd()) {
   path <- file.path(dir, db_name)
   con <- DBI::dbConnect(RSQLite::SQLite(), path)
+  if (DBI::dbExistsTable(con, "people")) {
```

Comment:
is this because of modifying test data in place? I think you'd be better off in the long creating things freshly, or you'll get weird test failures that depend on the order tests are run in

Reply:
So this wasn't a problem in the tests, everywhere in the tests I'm cleaning the database up using `on.exit(unlink(path_to_db))`. 

It was causing problem when knitting the vignette as the `people` table I'm creating in this setup would already exist. I'm not fully sure why it was failing to clear up the file when running in vignette. I see the point though, I can make this method just create the whole test db from new.